src/Domain/ComputationDomainMetric.hpp

```diff
@@ -0,0 +1,87 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace domain {
+
+/*!
+ * \brief The computation domain metric \f$\sum_k \frac{\partial x^k}{\partial
+ * \xi^i} \frac{\partial x^k}{\partial \xi^j}\f$, which is the flat spatial
+ * metric in element-logical coordinates.
+ *
+ * We define the "computation domain metric" to be the inner product of the i-th
+ * basis vector with the j-th basis vector, where the basis vectors are
+ * \f$\partial_\xi\f$, \f$\partial_\eta\f$, and \f$\partial_\zeta\f$. They can
+ * be expressed in the \f$x\f$-coordinate system, which gives a column of the
+ * Jacobian. Multiplying with the transpose from the left gives a matrix of the
+ * inner products of these basis vectors - the computation domain metric. It is
+ * just the spatial metric with respect to the logical coordinates, and we're
+ * assuming a flat metric here for now.
+ *
+ * When we need it, the computation domain metric in curved space can be
+ * computed just by a coordinate transformation of the spatial metric to logical
+ * coordinates, e.g. using `::transform::to_different_frame`.
+ */
+template <size_t Dim>
+void computation_domain_metric(
+    const gsl::not_null<tnsr::ii<DataVector, Dim, Frame::ElementLogical>*>
+        result,
+    const Jacobian<DataVector, Dim, Frame::ElementLogical, Frame::Inertial>&
+        jacobian) {
+  destructive_resize_components(result, jacobian.begin()->size());
+  for (size_t i = 0; i < Dim; ++i) {
+    for (size_t j = 0; j <= i; ++j) {
+      result->get(i, j) = 0.;
+      for (size_t k = 0; k < Dim; ++k) {
+        result->get(i, j) += jacobian.get(k, i) * jacobian.get(k, j);
```

Comment:
(This comment is more for myself that the math checks out here):

The first index (`k` here) indexes into the cartesian coordinates (x, y, z), while the second index (`i` and `j`) of the Jacobian indexes into the logical coordinates (xi, eta, zeta), so this line takes the cartesian dot product of the i-th logical coordinate basis vector with the j-th logical coordinate basis vector, which is what we wanted :slightly_smiling_face: